File: js/images.js
// images.js
const { dumpSchema } = require("./utils/index.js");
const BraveAPI = require("./API.js");

// Create a new instance of the BraveAPI class
const brave = new BraveAPI({
    subscription_token: process.env.BRAVE_KEY
});

// Setup the search parameters
const params = {
    q: "San Francisco Golden Gate Bridge",
    safesearch: "strict",
    count: 10,
};

// Perform the search
brave.images(params).then(results => {
    // Log the results
    console.log(dumpSchema(results));
});

File: python/images.py
# images.py
import os
from API import BraveAPI
from utils.main import dump_schema

# Create an instance of the BraveAPI class
brave = BraveAPI({
    'subscription_token': os.environ['BRAVE_KEY']
})

# Setting up parameters
params = {
    'q': "San Francisco Golden Gate Bridge",
    'safesearch': "strict",
    'count': 10
}

# Performing the images search
results = brave.images(params)

# Dumping and printing the schema of the results
print(dump_schema(results))
